iceoryx_posh/source/popo/ports/application_port.cpp

```diff
@@ -23,20 +24,27 @@ ApplicationPort::ApplicationPort(ApplicationPortData* const applicationPortDataP
 {
 }
 
-bool ApplicationPort::dispatchCaProMessage(const capro::CaproMessage& message) noexcept
+cxx::optional<capro::CaproMessage> ApplicationPort::getCaProMessage() noexcept
 {
-    return getMembers()->m_caproMessageFiFo.push(message);
+    auto maybeMessage = getMembers()->m_caproMessageFiFo.pop();
+
+    if (maybeMessage.has_value())
+    {
+        return cxx::make_optional<capro::CaproMessage>(maybeMessage.value());
+    }
+    else
+    {
+        return cxx::nullopt_t();
+    }
 }
 
-bool ApplicationPort::getCaProMessage(capro::CaproMessage& message) noexcept
+void ApplicationPort::dispatchCaProMessage(const capro::CaproMessage& caProMessage) noexcept
```

Comment:
caProMessage looks ugly for consistency i would take msg here. But i have not strong feelings about it as @elBoberido would say ;-)